File: P4/civitasruby/lib/civitas/casilla_impuesto.rb
#encoding:utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas

  class Casilla_impuesto < Casilla

    def initialize(importe, nombre)
      super(nombre)
      @importe = importe # debe ser positivo porque en paga (jugador) lo multiplica por *-1
    end
    
    @override
    def to_s
      "#{super}   Importe: #{@importe}"
    end

    def recibe_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        super
        todos.at(actual).paga_impuesto(@importe)
      end
    end

  end
  
end

File: Parcial2/Civitas/src/civitas/CasillaImpuesto.java
package civitas;

import java.util.ArrayList;

/**
 *
 * @author salva
 */
public class CasillaImpuesto extends Casilla{
    private float importe; //debe ser positivo porque en "paga" de jugador se multiplica por -1
    
    CasillaImpuesto(float importe, String nombre){
        super(nombre);
        this.importe = importe;
    }
    
    @Override
    public String toString(){
        String info = super.toString();
        info += "\nImporte del impuesto: " + importe;
        return info;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).pagaImpuesto(importe);
        }
    }
    
}
